README.md

```diff
@@ -7,209 +7,119 @@
-name you will use in the `-p` parameter later on.
+```
+bundle install
+```
 
-To see the full product list visit the [`products`](products/) folder.
 
+### Generating downstream tools
 
-## (One Time) Setup
+Before making any changes, you can compile the "downstream" tool you're working
+on by running the following command. If Magic Modules has been installed
+correctly, you'll get no errors when you run a command:
 
-Magic Modules requires:
+```
+bundle exec compiler -a -v "ga" -e {{tool}} -o "{{output_folder}}"
+```
 
-- git (so you can checkout the code)
-- Ruby 2.5
-- Bundler gem
+Generally, you'll want to generate into the same output; here's a reference of
+common commands
 
-### Ruby
+{{tool}}  | {{output_folder}}
+----------|----------
+terraform | $GOPATH/src/github.com/terraform-providers/terraform-provider-google
+ansible   | build/ansible
+inspec    | build/inspec
 
-Since Ruby 2.5 is not yet available in some common distributions, we recommend using
-`rbenv` to ensure that you are using the correct Ruby version.  If you see an error
-about `unexpected keyword_rescue`, that means you're using an older Ruby.
+For example, to generate Terraform:
 
-### Downloading code
+```
+bundle exec compiler -a -v "ga" -e terraform -o "$GOPATH/src/github.com/terraform-providers/terraform-provider-google"
+```
 
-Depending on the product and provider combination the generated code is stored
-on another Git repository. To make it easier to track them all these foreign Git
-repositories are tracked as submodules of Magic Modules main repository.
+It's worth noting that Magic Modules will only generate new files when ran
+locally. The "Magician"- the Magic Modules CI system- handles deletion of old
+files when creating PRs.
 
-To fetch all products last known good build, run:
+#### Terraform's `google-beta` provider
 
-    git submodule update --init
+Terraform is the only tool to handle Beta features right now; you can generate
+`google-beta` by running the following, substitution `"beta"` for the version
+and using the repository for the `google-beta` provider.
 
-### Installing bundler
+```
+bundle exec compiler -a -v "beta" -e terraform -o "$GOPATH/src/github.com/terraform-providers/terraform-provider-google-beta"
+```
 
-As you have Ruby installed all you need is Bundler. To install it, run:
+### Testing your changes
 
-    gem install bundler
+Once you've generated your changes for the tool, you can test them by running the
+tool-specific tests as if you were submitting a PR against that tool.
 
-### Dependencies install
+You can run tests in the `{{output_folder}}` from above. See the following for
+more details;
 
-Now that we have Bundler installed, go to the root folder where you checked out
-the Magic Modules code and run:
+Tool             | Testing Guide
+-----------------|--------------
+ansible          | [instructions](https://docs.ansible.com/ansible/devel/dev_guide/testing.html)
```

Comment:
I feel like we might want to put our own instructions, but this is a good starting point.

Reply:
My intention here is that we just defer to the downstreams; users should read how to test Ansible, or Inspec, or Terraform, and not how to use MM to test them.

Reply:
I'll leave it to you and @slevenick to fill in whatever you want for your tools afterwards, but at least for Terraform the entire story right now is "test it like it's handwritten code"